src/test/fio/fio_ceph_messenger.cc

```diff
@@ -80,7 +80,7 @@ struct ceph_msgr_reply_io {
 
 static void *str_to_ptr(const std::string &str)
 {
-  return (void *)strtoul(str.c_str(), NULL, 16);
+  return (void *)strtoull(str.c_str(), NULL, 16);
```

Comment:
how about be more explicit by using 
```c++
return reinterpret_cast<void*>(ceph::parse<uintptr_t>(str, 16));
```
?

Reply:
it's really a explicit way, but we should use
`reinterpret_cast<void*>(ceph::parse<uintptr_t>(str, 16).value());`

Reply:
I reviewed @sleepinging's original PR (https://github.com/ceph/ceph/pull/45682) and I wasn't aware of `ceph::parse`.  Looking at `ceph::parse` template, it would return `std::optional<uintptr_t>` in this case.  This means that an empty optional would need to be handled.

Would that `reinterpret_cast` even compile here?

Reply:
Sorry, my comment was with regard to @tchaikov's suggested change.

Appending `.value()` would fix the compilation issue but `str_to_ptr` would now crash due to unhandled exception on invalid input.

Reply:
Yes, I think we can but it's probably worth a comment on the line before `reinterpret_cast`.